iroha-cli/interactive/impl/interactive_query_cli.cpp

```diff
@@ -160,11 +164,54 @@ namespace iroha_cli {
           local_time_, creator_, counter_, account_id, asset_id);
     }
 
+    nonstd::optional<iroha::model::Pager> InteractiveQueryCli::parsePager(
+        const std::string& encoded_tx_hash, const std::string& limit_str) {
+      return nonstd::make_optional<iroha::model::Pager>()
+        | [&encoded_tx_hash](iroha::model::Pager pager)
+            -> nonstd::optional<iroha::model::Pager> {
+            /**
+             * Use '-' as the argument of tx_hash to specify that
+             * the hash is empty because interactive cli splits
+             * arguments by space.
+             */
+            const auto PAGER_TX_HASH_EMPTY = "-";
+            if (encoded_tx_hash == PAGER_TX_HASH_EMPTY) {
+              pager.tx_hash.fill(0);
+            } else {
+              const auto decoded_hash =
+                  iroha::hexstringToBytestring(encoded_tx_hash);
+              if (not decoded_hash) {
+                return nonstd::nullopt;
+              }
+              pager.tx_hash = iroha::hash256_t::from_string(*decoded_hash);
+            }
+            return pager;
+          }
+        | [&limit_str](auto pager)
+            -> nonstd::optional<iroha::model::Pager> {
+          try {
+            pager.limit = boost::lexical_cast<decltype(pager.limit)>(limit_str);
+          } catch (...) {
+            return nonstd::nullopt;
+          }
+          return pager;
+        };
+    }
+
     std::shared_ptr<iroha::model::Query>
     InteractiveQueryCli::parseGetAccountTransactions(QueryParams params) {
-      auto account_id = params[0];
-      return generator_.generateGetAccountTransactions(
-          local_time_, creator_, counter_, account_id);
+      /// TODO 01/11/17 motxx - Using cli with environment
+      /// options can remove unnamed arguments in interactive-cli.
+      /// Example: GetAccountTransactions with no tx_hash
+      /// ./iroha-cli GetAccountTransactions --from alice@domain
+      ///             --to alice@domain --limit 10
+      return (parsePager(params[1], params[2]) |
```

Comment:
Bracket (before `parsePager`) looks redundant

Reply:
Cannot delete this bracket.
```cpp
(parsePager...).value_or(nullptr);
```

Is it better that
```cpp
auto p = parsePager() | ...;
return p.value_or(nullptr);
```
?

Reply:
Personally I like more the last one